web/widgets.py

```
@@ -135,3 +139,224 @@ def __init__(self, attrs=None):
+                           class="w-full border-none p-1 focus:outline-none focus:ring-1
+                                  focus:ring-teal-300 bg-transparent" />
+                </div>
+                <button type="button"
+                        class="ml-2 text-red-500 hover:text-red-700 dark:text-red-400 dark:hover:text-red-300
+                               transition-colors duration-200 delete-btn">
+                    <i class="fas fa-trash-alt"></i>
+                </button>
+            </div>
+            """.format(
+                i=i, escaped_feature=html.escape(str(feature))
+            )
```

Ruff (0.8.2)

Use f-string instead of `format` call

Convert to f-string

(UP032)